### ISSUE TITLE
sails-postgresql .select() does not exist

### ISSUE THREAD

**Issue by billyshena** [balderdashy/waterline#1331]
**Waterline version**: 0.12.1
**Node version**: 5.10
**NPM version**: 3.3.13
**Operating system**: Mac OSX

<hr/>

I have recently updated my waterline package to the latest 0.12.1 version to be able to use .select() projection methods. However, when I try a simple Model.find().select(...), it says that .select() is not a function.
Should I update my waterline-sequel hook aswell?

Best,

**Comment by sailsbot** [comment]
Hi @billyshena!  It looks like you missed a step or two when you created your issue.  Please edit your comment (use the pencil icon at the top-right corner of the comment box) and fix the following:

* Verify "I am experiencing a concrete technical issue with Waterline (ideas and feature proposals should follow the guide for proposing features and enhancements (http://bit.ly/sails-feature-guide), which involves making a pull request).  If you're not 100% certain whether it's a bug or not, that's okay--you may continue.  The worst that can happen is that the issue will be closed and we'll point you in the right direction."
* Verify "I am not asking a question about how to use Waterline or about whether or not Waterline has a certain feature (please refer to the documentation(http://sailsjs.org), or post on http://stackoverflow.com, our Google Group (http://bit.ly/sails-google-group) or our live chat (https://gitter.im/balderdashy/sails)."
* Verify "I have already searched for related issues, and found none open (if you found a related _closed_ issue, please link to it in your post)."
* Verify "My issue title is concise, on-topic and polite."
* Verify "**I can provide steps to reproduce this issue that others can follow.**"

As soon as those items are rectified, post a new comment (e.g. &ldquo;Ok, fixed!&rdquo;) below and we'll take a look.  Thanks!

*If you feel this message is in error, or you want to debate the merits of my existence (sniffle), please contact inquiries@treeline.io.*

**Comment by particlebanana** [comment]
@billyshena can you create a repo that reproduces the issue? The `.select()` method exists in the current version of Waterline (0.12.x) [here](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/deferred.js#L304) and is tested.

**Comment by billyshena** [comment]
@particlebanana: Hi Particlebanana ! Thanks for your reply
I'm currently trying to update the waterline package since the default version of the package generated by sails 0.12.x is the 0.11.x version.
Is there a way to update the waterline package using the root package.json of my Sails project instead of overriding the version in node_modules/sails/waterline/package.json ?

Best,

**Comment by particlebanana** [comment]
@billyshena I'm not sure if Sails `0.12.x` supports the latest Waterline.

**Comment by billyshena** [comment]
@particlebanana Yea thought so. What would be the best way to update my waterline package to the 0.12.x version then?
Is there a way to directly specify the waterline version in the root package.json of my Sails project?

**Comment by sgress454** [comment]
> Is there a way to directly specify the waterline version in the root package.json of my Sails project?

There is not (and for good reason).  We're working now on getting Sails to be Waterline 0.12 compatible, we'll post back here when it's done!

**Comment by billyshena** [comment]
@sgress454 @particlebanana Looking forward for the new update then :) !
Thanks for your support

**Comment by sailsbot** [comment]
@billyshena,@sailsbot,@particlebanana,@sgress454: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!
